uber/site_sections/groups.py

```diff
@@ -1,6 +1,37 @@
 from uber.common import *
 
 
+def _decline_and_convert_dealer_group(session, group, convert=False):
+    message = 'Group declined'
+    for attendee in group.attendees:
+        if attendee.is_unassigned or not convert:
+            session.delete(attendee)
+        else:
```

Comment:
We're probably going to need a few checks here before we do anything to a badge.  In particular, there were a few weird edge cases where people were in the system attached to unapproved dealer registrations and then some admin ended up making them a guest or panelist or MIVS judge.  Presumably the admins were like, "Oh cool this person is already in the database, lemme just mark them appropriately."

Last year this led to some people being deleted who shouldn't have.  In this case that wouldn't be as bad, but if someone was e.g. marked as having a ``c.NEED_NOT_PAY`` badge by an admin after they became a Panelist or something then we wouldn't want to change that.

The problem is that I'm not sure that there's an easy set of checks we can do to ensure that this is someone who we should be changing.  Off the top of my head:
- The only people in these groups should be the group leaders, since any normal unapproved dealer would only have the leader.  So if a group has more than one un-unassigned attendee in it then something is up and some admin must have done something.
- The attendee should have PAID_BY_GROUP Attendee badge with a dealer ribbon.
- The attendee and no admin account.
- The attendee should not be signed up for any volunteer shifts.

If any of the above things are true, then we should either disassociate the attendee from the group without changing the attendee OR just not touch any waitlisted dealer which violates any of these rules.

I think that would check for everything we hit last year.

Reply:
I know that some dealers set up their assistants while waiting to hear from us, plus all the cases Eli mentioned.

I'm wondering why we would delete anything other than an Unassigned badge.